go/vt/sqlparser/ast_funcs.go

```diff
@@ -764,6 +771,52 @@ func (node *Select) MakeDistinct() {
 	node.Distinct = true
 }
 
+// ContainsOnlyInformationSchema returns if the select statement only contains information_schema tables
+func (node *Select) ContainsOnlyInformationSchema() bool {
+	for _, expr := range node.From {
+		if !expr.ContainsOnlyInformationSchema() {
+			return false
+		}
+	}
+	return true
+}
+
+// ContainsOnlyInformationSchema returns if the expression only contains information_schema tables
+func (expr *ParenTableExpr) ContainsOnlyInformationSchema() bool {
+	for _, tableExpr := range expr.Exprs {
+		if !tableExpr.ContainsOnlyInformationSchema() {
+			return false
+		}
+	}
+	return true
+}
+
+// ContainsOnlyInformationSchema returns if the expression only contains information_schema tables
+func (expr *JoinTableExpr) ContainsOnlyInformationSchema() bool {
+	if !expr.LeftExpr.ContainsOnlyInformationSchema() {
+		return false
+	}
+	if !expr.RightExpr.ContainsOnlyInformationSchema() {
+		return false
+	}
+	return true
```

Comment:
nit: can be re-written as `return expr.LeftExpr.ContainsOnlyInformationSchema() & expr.RightExpr.ContainsOnlyInformationSchema()`

Reply:
Done